### DISCUSSION TITLE
Angular-query: When is the end date for the experimental version of Angular Query?

### DISCUSSION THREAD

**Discussion by load28** [TanStack/query#7144] [Q&A] [answered]
When is the end date for the experimental version of Angular Query?
I want to integrate Angular Query into my product.

**Comment by arnoud-dv** [comment, marked as answer]
I cannot give an end date. I work on this in my spare time of which I didn't have much lately.

There is an issue with type narrowing on query results in templates. I hope to test improvements on signal type narrowing in Angular 18 soon and see if this will be solved without needing API changes in the adapter.

Contributions and testing on injectQueries and SSR would help. I am working on adding observable support to queryFn and I hope to finish this in a week or two.

**Reply by load28** [reply]
Thank you! 
I'll test out various things and provide feedback to help ensure that Angular Query gets officially released!

**Reply by aberonni** [reply]
Thanks for your work on this, it's much appreciated! ❤️ I'm wondering if you have a better idea on an ETA since this comment was posted and v18 has been released.

**Reply by driatvak** [reply]
Great to know that you're working on that !! 
Any idea about when the experimental will be over ? 🙏

**Comment by ExFlo** [comment]
Hi see [PR type narrowing](https://github.com/TanStack/query/pull/7923) it it may helps @arnoud-dv

**Comment by rbalet** [comment]
Hi @arnoud-dv, could I kindly ask about the status of this discussion, what are for you what's blocking the release of a stable version ? 

I do believe the experimental warning on npm may lower the people adopting this library and would love to see it growing.

Thanks again for the work done

**Reply by arnoud-dv** [reply]
Hi Raphaël, you can follow what needs to be done in [this issue](https://github.com/TanStack/query/issues/8703)

- [I've made a PR](https://github.com/TanStack/query/pull/9270) that fixes most of the open issues. Ran into problems with the publish script though. I'll do some more testing and hopefully merge it soon.

- Unless somebody can complete [the PR for injectQueries](https://github.com/TanStack/query/issues/8704) I'll remove that function for now and work on it after stable.
